templates/plugin/container_gateway.yml.erb

```
@@ -6,3 +6,19 @@
 <% if scope.lookupvar("foreman_proxy::plugin::container_gateway::sqlite_timeout") -%>
 :sqlite_timeout: <%= scope.lookupvar("foreman_proxy::plugin::container_gateway::sqlite_timeout") %>
 <% end -%>
+<% if scope.lookupvar("foreman_proxy::plugin::container_gateway::database_backend") == 'postgres' -%>
+:db_connection_string: <%=
+  "#{scope.lookupvar("foreman_proxy::plugin::container_gateway::database_backend")}://" \
+  "#{scope.lookupvar("foreman_proxy::plugin::container_gateway::postgresql_user")}:" \
+  "#{scope.lookupvar("foreman_proxy::plugin::container_gateway::postgresql_password")}@" \
+  "#{scope.lookupvar("foreman_proxy::plugin::container_gateway::postgresql_host")}:" \
+  "#{scope.lookupvar("foreman_proxy::plugin::container_gateway::postgresql_port")}/" \
+  "#{scope.lookupvar("foreman_proxy::plugin::container_gateway::postgresql_database")}"
+  %>
+<% end -%>
```

```suggestion
<% if scope.lookupvar('foreman_proxy::plugin::container_gateway::database_backend') == 'postgres' -%>
:db_connection_string: <%=
  [
    scope.lookupvar('foreman_proxy::plugin::container_gateway::database_backend') + '://',
    scope.lookupvar('foreman_proxy::plugin::container_gateway::postgresql_user') + ':',
    scope.lookupvar('foreman_proxy::plugin::container_gateway::postgresql_password') + '@',
    scope.lookupvar('foreman_proxy::plugin::container_gateway::postgresql_host') + ':',
    scope.lookupvar('foreman_proxy::plugin::container_gateway::postgresql_port') + '/',
    scope.lookupvar('foreman_proxy::plugin::container_gateway::postgresql_database')
  ].join
  %>
<% end -%>
```

Untested, but maybe this approach is a bit nicer?

---

Gonna revert this, seeing ` Detail: String can't be coerced into Integer`

---

Ah, Stdlib::Port is an Integer type: https://github.com/puppetlabs/puppetlabs-stdlib/blob/main/types/port.pp#L2

I expect it would work if you tried `scope.lookupvar('foreman_proxy::plugin::container_gateway::postgresql_port').to_s + '/',` instead

If you don't wanna go that route, I'd recommend at least using single quotes for the `lookupvar` args... not sure if it can handle the nested double quotes without escaping.... and that's the type of thing where I prefer not needing to know, lol

---

Another way you could consider is

```
<% if scope.lookupvar('foreman_proxy::plugin::container_gateway::database_backend') == 'postgres' -%>
:db_connection_string: <%= [
  scope.lookupvar('foreman_proxy::plugin::container_gateway::database_backend'),
  '://',
  scope.lookupvar('foreman_proxy::plugin::container_gateway::postgresql_user'),
  ':',
  scope.lookupvar('foreman_proxy::plugin::container_gateway::postgresql_password'),
  '@',
  scope.lookupvar('foreman_proxy::plugin::container_gateway::postgresql_host'),
  ':',
  scope.lookupvar('foreman_proxy::plugin::container_gateway::postgresql_port'),
  '/',
  scope.lookupvar('foreman_proxy::plugin::container_gateway::postgresql_database')
].join %>
<% end -%>
```

---

I like the second version you posted, I'll use that.

---

> not sure if it can handle the nested double quotes without escaping.... and that's the type of thing where I prefer not needing to know, lol

It apparently can, cuz it worked! But yeah I'm up for avoiding that.